### PR TITLE
Reset iteration budget and update default max_iterations to 500

### PR DESCRIPTION
**Pull request by rbren**
This PR implements two changes to improve iteration handling:

1. Updates the default max_iterations to 500 in core config
2. Resets iteration budget when the agent pauses for user interaction

Changes:
- Modified OH_MAX_ITERATIONS in config_utils.py to 500
- Added iteration budget reset in set_agent_state_to() when agent enters PAUSED, AWAITING_USER_INPUT, or AWAITING_USER_CONFIRMATION states

The agent will now get a fresh budget of max_iterations whenever it pauses for user interaction, allowing it to continue for up to max_iterations more steps after each user input. This ensures the agent can make progress on complex tasks that require multiple user interactions while still maintaining control over continuous execution.

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:308d1ac-nikolaik   --name openhands-app-308d1ac   docker.all-hands.dev/all-hands-ai/openhands:308d1ac
```

### CODE REVIEW

File: openhands/controller/agent_controller.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def __init__(
         self,
         agent: Agent,
         event_stream: EventStream,
-        max_iterations: int,
+        max_iterations: int = 250,
```
Comment (rbren) [inline openhands/controller/agent_controller.py:82]:
```suggestion
        max_iterations: int,
```

File: openhands/controller/agent_controller.py [hunk 2]
```diff
@@ -339,6 +339,9 @@ async def set_agent_state_to(self, new_state: AgentState) -> None:
 
         if new_state in (AgentState.STOPPED, AgentState.ERROR):
             self._reset()
+        elif new_state in (AgentState.PAUSED, AgentState.AWAITING_USER_INPUT, AgentState.AWAITING_USER_CONFIRMATION):
+            # Reset iteration budget when agent pauses for user interaction
```
Comment (enyst) [inline openhands/controller/agent_controller.py:343]: There might be a problem, basically the same as the other PR on this: evals also use a user message, when the agent happens to be waiting for input - I think the state change does happen

https://github.com/All-Hands-AI/OpenHands/pull/5560

Comment (rbren) [reply openhands/controller/agent_controller.py:343]: ah dang let's revert for now

File: openhands/controller/agent_controller.py [hunk 3]
```diff
@@ -339,6 +339,9 @@ async def set_agent_state_to(self, new_state: AgentState) -> None:
...
+            self.state.max_iterations = self.state.iteration + self._initial_max_iterations
```
Comment (rbren) [inline openhands/controller/agent_controller.py:344]:
```suggestion
```

**Review by enyst — APPROVED** [review]

**Comment by enyst** [comment]
Thank you! 🥳

**Pull request merged**